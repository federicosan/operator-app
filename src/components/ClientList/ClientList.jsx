import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import ClientCard from '../ClientCard/ClientCard.jsx';
import './ClientList.css';


const ClientList = (props) => {
  const { operatorFilter, chats } = props;

  // filters by operator filter + maps to html
  function getChats () {
    const filtered = chats.filter((chat) => {
      if (operatorFilter === 'open') return chat.open;
      if (operatorFilter === 'closed') return !chat.open;
      return chat;
    });

    return filtered.map(chat => (
      <ClientCard key={chat.id} chatId={chat.id}>
        {`${chat.client.first_name} ${chat.client.last_name}`}
      </ClientCard>
    ));
  }


  return <ul className="ClientList__list">{ getChats() }</ul>;
};

const mapStateToProps = state => ({
  operatorFilter: state.chat.operatorFilter,
});

const mapDispatchToProps = dispatch => ({

});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ClientList);


ClientList.propTypes = {
  chats: PropTypes.array.isRequired,
  operatorFilter: PropTypes.string.isRequired,
};
