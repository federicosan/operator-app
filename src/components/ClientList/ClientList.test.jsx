import React from 'react';

import { shallow } from 'enzyme';

import ClientList from './ClientList.jsx';

const store = {
  subscribe: jest.fn(),
  dispatch: jest.fn(),
  getState: jest.fn(() => ({
    chat: {
      operatorFilter: 'all',
    },
  })),
};


describe('ClientList', () => {
  const chats = [
    { client: { first_name: 'Robert', last_name: 'waffle' } },
    { client: { first_name: 'Lisa', last_name: 'pancake' } },
  ];

  it('matches snapshot', () => {
    const component = shallow(<ClientList chats={chats}store={store} />);
    expect(component).toMatchSnapshot();
  });
});
