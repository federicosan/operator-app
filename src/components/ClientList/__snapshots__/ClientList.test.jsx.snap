exports[`ClientList matches snapshot 1`] = `
<ClientList
  chats={
    Array [
      Object {
        "client": Object {
          "first_name": "Robert",
          "last_name": "waffle",
        },
      },
      Object {
        "client": Object {
          "first_name": "Lisa",
          "last_name": "pancake",
        },
      },
    ]
  }
  operatorFilter="all"
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  } />
`;
